packages/block-library/src/template-part/edit/index.js

```diff
@@ -63,9 +68,18 @@ export default function TemplatePartEdit( {
 	);
 
 	const blockProps = useBlockProps();
-	const isPlaceholder = ! slug;
-	const isEntityAvailable = ! isPlaceholder && ! isMissing;
 
+	if ( hasAlreadyRendered ) {
```

Comment:
Since `templatePartId` might be `null`. In theory, we should move this check after `isMissing` verification. Although in practice it should never be an issue because it bails out early. Anyway, for the sake of correctness, I will update the flow.